tests/test_show_git_status_complex_cases.py

```
@@ -0,0 +1,37 @@
+
+            mock_run.side_effect = side_effect
+
+            with patch("sys.stdout", new=StringIO()) as fake_out:
+                git_commitai.show_git_status()
+                output = fake_out.getvalue()
+                assert "HEAD detached at abc1234" in output
+
+    def test_show_git_status_all_exceptions(self):
+        """Test show_git_status when all git commands fail."""
+        with patch("git_commitai.run_git", side_effect=Exception("Git completely broken")):
+            with patch("sys.stdout", new=StringIO()) as fake_out:
```

Ruff (0.12.2)

Use a single `with` statement with multiple contexts instead of nested `with` statements

(SIM117)